ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -454,6 +454,9 @@ private TopicDescription getTopicDescription(final Admin admin, final String sou
         new ListOffsetsOptions(IsolationLevel.READ_UNCOMMITTED)
     );
 
+    final ImmutableMap<TopicPartition, Long> startOffsetsForStreamPullQuery =
```

Comment:
Sigh.. The `ListOffsetsPartition` in the request can support fetching for multiple specs actually, but it seems they are not exposed via admin APIs. But I guess paying one more round trip is not too much of a deal breaker for stream pull queries.